types/mempool_test.go

```diff
@@ -0,0 +1,179 @@
+package types_test
+
+import (
+	"crypto/sha256"
+	"fmt"
+	simtestutil "github.com/cosmos/cosmos-sdk/testutil/sims"
+	simtypes "github.com/cosmos/cosmos-sdk/types/simulation"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	"github.com/cosmos/cosmos-sdk/x/group"
+	"github.com/stretchr/testify/require"
+	"math/rand"
+
+	moduletestutil "github.com/cosmos/cosmos-sdk/types/module/testutil"
+
+	"github.com/cosmos/cosmos-sdk/types"
+	"github.com/tendermint/tendermint/libs/log"
+
+	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
+	"testing"
+)
+
+var (
+	smallSize = 10
+	coinDenom = "aCoin"
+	acc       = authtypes.NewEmptyModuleAccount("anaccount")
+)
+
+type ConfigurableBenchmarker struct {
+	Name     string
+	GetBench func(input []byte) func(b *testing.B)
+}
+
+func TestInsertMemPool(t *testing.T) {
+	mPool := types.NewBTreeMempool(smallSize)
+	ctx := types.NewContext(nil, tmproto.Header{}, false, log.NewNopLogger())
+	tx := simulateTx(ctx)
+	err := mPool.Insert(ctx, TxWSize{tx})
+	require.NoError(t, err)
+}
+
+func TestSelectMempool(t *testing.T) {
+	maxBytes := 10
+	mPool := types.NewBTreeMempool(smallSize)
+	ctx := types.NewContext(nil, tmproto.Header{}, false, log.NewNopLogger())
+	transactions := simulateManyTx(ctx, 1000)
+	for _, t := range transactions {
+		mPool.Insert(ctx, TxWSize{t})
+	}
+	selectedTx, err := mPool.Select(ctx, nil, maxBytes)
+	require.NoError(t, err)
+	actualBytes := 0
+	for _, selectedTx := range selectedTx {
+		actualBytes += selectedTx.Size()
+	}
+	require.LessOrEqual(t, maxBytes, actualBytes)
+
+}
+
+func BenchmarkBtreeMempool_Insert(b *testing.B) {
```

Comment:
can we use table tests to remove a bunch of boilerplate?

Reply:
Sure sure this was just to start testing I will clean up quite a bit